### DISCUSSION TITLE
Alarm rootfs (/): low disk space

### DISCUSSION THREAD

**Discussion by 4EverChaos** [OpenCCU/OpenCCU#2664] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Erhalte folgende Fehlermeldung

![ksnip_20240221-210119](https://github.com/jens-maus/RaspberryMatic/assets/20836164/96069124-bf51-4c8c-b8cb-f283325593fc)

Unter Hilfe wird der Free Space wie folgt gelistet
rootfs Free Space: 0 B [0.0%] / 973.4 MB

Wie kann ich den Speicher freigegeben? Ich habe bereits ein System-Reset durchgeführt, wie [hier](https://github.com/jens-maus/RaspberryMatic/discussions/2161) beschrieben - allerdings ohne Erfolg.

### Describe the behavior you expected

Mehr freier Speicherplatz verfügbar. 
Erlischen der Alarmmeldung.

### Steps to reproduce the issue

Unklar warum der Speicher voll gelaufen ist -> wird auch per System-Reset nicht freigegeben.

### What is the version this bug report is based on?

3.73.9.20240130

### Which base platform are you running?

rpi3 (RaspberryPi3)

### Which HomeMatic/homematicIP radio module are you using?

n/a

### Anything in the logs that might be useful for us?

```txt
Welche .log Datei kann hier behilflich sein?
```

**Comment by Baxxy13** [comment]
> wird auch per System-Reset nicht freigegeben.

Bei einem System-Reset wir nur das /userfs "bereinigt".
Das /rootfs wird dabei nicht angefasst.

> Unklar warum der Speicher voll gelaufen ist

Das kannst du nur selber analysieren. 
Normalerweise ist das /rootfs als "read-only" gemounted und kann daher nicht beschrieben werden.

**Reply by 4EverChaos** [reply]
Welche Mittel stehen zur Verfügung um das selbst zu analysieren?

**Reply by jp112sdl** [reply]
Was kommt denn bei `find / -type d -maxdepth 1 -exec du -sh {} \;` raus?

Edit:
Falls du nicht per SSH drauf kommst, dann geht es notfalls auch mit "Skript testen":
```
string a;string b;
system.Exec('find / -type d -maxdepth 1 -exec du -sh {} \;',&a,&b);
WriteLine(a);
WriteLine(b);
```
Aber nicht ungeduldig werden. Das läuft nen Augenblick...

**Reply by 4EverChaos** [reply]
> Was kommt denn bei find / -type d -maxdepth 1 -exec du -sh {} \; raus?

![ksnip_20240221-214834](https://github.com/jens-maus/RaspberryMatic/assets/20836164/39f27770-f543-41c1-81b5-3676efbee273)

**Reply by jens-maus** [reply]
@jp112sdl 
> Was kommt denn bei `find / -type d -maxdepth 1 -exec du -sh {} \;` raus?

Warum denn mit `find` und so kompliziert? Das kann `du` doch von Haus aus via

```
du -hx -d 1 /
```

😜

**Reply by jp112sdl** [reply]
Wenn ich `/usr` und `/boot` von den 1.4G abziehe komme ich bei Weitem nicht auf 973 MB.

Aber auffällig ist, dass `/www` sehr groß ist. 

Check mal noch `du -hx -d 1 /usr` und `du -hx -d 1 /www`

**Reply by 4EverChaos** [reply]
Danke für die Befehle, die waren mir bisher nicht so bekannt.
Es sieht dann wie folgt aus...
<img width="287" alt="Bildschirmfoto 2024-02-22 um 18 46 59" src="https://github.com/jens-maus/RaspberryMatic/assets/20836164/085e3688-0373-411b-a076-93cac3a16cb8">

Mir ist eingefallen dass noch eine Instanz für RedMatic (für die Homekit Integration) auf dem CCU3 mitläuft, aber ansonsten fallen mir keine extra installierten/konfigurierten oder laufenden Instanzen ein.

**Reply by jens-maus** [reply]
irgendwas hast du direkt unter `/www` liegen was da nicht hingehört und deshalb insgesamt 230MB verbraucht. Liste mal den content von `/www` auf um rauszufinden was das sein soll. Einfach via:

```
ls -laSr /www
```

Um den inhalt größensortiert anzuzeigen.

**Reply by 4EverChaos** [reply]
Anbei die Ausgabe mit dem ls Befehl.

![ksnip_20240222-214622](https://github.com/jens-maus/RaspberryMatic/assets/20836164/58d59069-24bc-40a4-b52a-e7daeae9ee1d)

**Reply by jens-maus** [reply, marked as answer]
> Anbei die Ausgabe mit dem ls Befehl.

Na da haben wir doch den Übeltäter. Diese `rega\Datei` hat da definitiv nichts verloren. Und wer/was diese dort unter /www angelegt hat (wohl letztmalig am 21.Feb) kannst wohl nur du beantworten bzw. In Erfahrung bringen. Irgendetwas scheint diese Datei auf deinem System unter dem rootfs anzulegen und das obwohl das rootfs eben aus gutem Grund read-only gemountet wird und man dieses explizit read-write remounten muss im daran Änderungen vorzunehmen.

Was aber klar ist: Diese Datei ist der Auslöser für die WatchDog-Meldungen und gehört gelöscht. Und wenn sie danach wieder auftaucht dann musst du mal auf deiner Zentrale selbst nachforschen wer/was diese anlegt. RaspberryMatic selbst ist/kann das nicht sein.

**Reply by 4EverChaos** [reply]
Leider waren meine Versuche die Datei zu entfernen bisher ohne Erfolg. 
Was merkwürdig ist, dass die Datei per Auto Vervollständigung gar nicht richtig erkannt wird. Wenn ich den Namen kopiere und direkt eingebe zum löschen, wird die nicht gefunden / existiert nicht. (No such file or directory)
Nur wenn ich es mit einem Platzhalter (*) voraus eingebe, komm ich in der Hinsicht weiter. Aber wenn ich soweit bin das Löschen mit yes zu bestätigen, folgt der Hinweis "Read-only file system". Was ich wegen /www nachvollziehen kann. 
Habt ihr eventuell einen Tipp wie ich das read-only umgehen kann?

![ksnip_20240223-172140](https://github.com/jens-maus/RaspberryMatic/assets/20836164/30e7cce1-a3da-4f5f-a739-03b7bc56c562)

**Reply by jp112sdl** [reply]
Wie Jens schon schrieb und es dort auch steht - das Dateisystem ist read-only gemountet.

```
mount -o remount,rw /
```
macht es beschreibbar.

**Reply by 4EverChaos** [reply]
Nach dem mount Befehl konnte ich die dubiose Datei entfernen und habe wieder freien Speicher auf rootfs. Somit wurde auch nicht mehr die Alarmmeldung ausgelöst. 

![ksnip_20240224-172403](https://github.com/jens-maus/RaspberryMatic/assets/20836164/fe3e4e33-ac14-45e1-8015-a4312b9a22da)

Aus meiner Sicht wäre das Problem damit behoben, danke an alle die mir weiter geholfen haben.

**Reply by jens-maus** [reply]
Das Problem ist vllt behoben, die Ursache aber nicht geklärt! Mach lieber mal ein frischen reboot oder erneutes FW Update und schaue ob die Datei irgendwann wieder auftaucht (was ich stark vermute!). Und dann muss geklärt werden wer konkret diese erzeugt!

**Reply by 4EverChaos** [reply]
Das ist richtig. Ein reboot hab ich gestern Abend direkt nach dem löschen durchgeführt, bis jetzt ist sie zumindest nicht wieder aufgetaucht. Es wird weiter beobachtet.

**Reply by jp112sdl** [reply]
Evtl. fällt dir ja noch ein, was du am 21.02. um 14:21 Uhr gemacht hast.
